vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/model/types/Type.kt

```diff
@@ -129,42 +139,57 @@ sealed interface Type {
     /**
      * A [Type] that represents a [Datetime] value.
      */
+    @Serializable
     data object Datetime : Type {
         override val dimensions: kotlin.Int = 1
         override fun defaultValue(): Value<*> = Value.DateTime(Date())
     }
 
+    /**
+     * A [Type] that represents a [UUID] value.
+     */
+    @Serializable
+    data object UUID : Type {
```

Comment:
Just so I can understand why this is needed: UUIDs (thus far) are only used internally by the engine as IDs. Adding this as a type basically means, that we want to be able to actually use UUIDs as fields. Is that the intent?

Reply:
Since I had to touch the type system anyway to make everything serializable, I thought adding this would be nice for completeness' sake. This way, the type can be used, for example, in a struct descriptor that needs to deal with external identifiers.